### DISCUSSION TITLE
batch processing

### DISCUSSION THREAD

**Discussion by lconcha** [kschan0214/sepia#80] [General]
Hi. Thanks a lot for this fantastic software.

I am able to go through all processing steps through the GUI. Now that I have all my images converted and in separate folders according to the SEPIA conventions, I would like to batch process them all using the same parameters. I cannot find a way to do it in the documentation, it is all GUI-based. Is this even possible in SEPIA?

**Comment by lconcha** [comment]
Well, answering my own question thanks to your well-documented code. 
I see that [the goal of the GUI is to create handy little .m files to be fed to sepiaIO](https://github.com/kschan0214/sepia/blob/3db9be46322f7269518b5f493562e2cce2eea6f2/sepia.m#L213). So it seems all I have to do is create one, and uset as a base in a for loop changing file names and keeping `algorParam` equal across evaluations. Awesome!

**Reply by kschan0214** [reply]
Yes! Once you have decided on the processing pipeline, you can run it on one dataset. This will automatically generate a .m script file that contains all method parameters and can be run without the GUI. You can then adopt this script for multiple datasets for example using a for-loop or using bash shell for automation.

**Reply by lconcha** [reply]
Fantastic, thanks for your reply and your great work. Marking this as closed.
